paradox-nakadi-consumer-boot/src/main/java/de/zalando/paradox/nakadi/consumer/boot/ControllerConfiguration.java

```diff
@@ -29,18 +26,85 @@
 
 import de.zalando.paradox.nakadi.consumer.boot.components.EventReceiverRegistry;
 import de.zalando.paradox.nakadi.consumer.boot.components.EventTypeConsumer;
+import de.zalando.paradox.nakadi.consumer.boot.components.FailedEventSourceNameResponse;
+import de.zalando.paradox.nakadi.consumer.boot.components.NumberOfFailedEventsResponse;
 import de.zalando.paradox.nakadi.consumer.core.EventHandler;
 import de.zalando.paradox.nakadi.consumer.core.client.Client;
 import de.zalando.paradox.nakadi.consumer.core.domain.EventType;
 import de.zalando.paradox.nakadi.consumer.core.domain.EventTypeCursor;
 import de.zalando.paradox.nakadi.consumer.core.domain.EventTypePartition;
+import de.zalando.paradox.nakadi.consumer.core.domain.FailedEvent;
 
 import rx.Single;
 
 @Configuration
 public class ControllerConfiguration {
 
     @RestController
+    @RequestMapping(value = "/nakadi/failed-event-sources")
+    public static class FailedEventReplayController {
+
+        private final FailedEventReplayer failedEventReplayer;
+
+        @Autowired
+        public FailedEventReplayController(final FailedEventReplayer failedEventReplayer) {
+            this.failedEventReplayer = failedEventReplayer;
+        }
+
+        @RequestMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+        public FailedEventSourceNameResponse getFailedEventSources() {
+            return new FailedEventSourceNameResponse(failedEventReplayer.getFailedEventSources());
+        }
+
+        @RequestMapping(value = "/{event_source_name:.+}", produces = MediaType.APPLICATION_JSON_VALUE)
+        public NumberOfFailedEventsResponse getTotalNumberOfFailedEvents(
+                @PathVariable("event_source_name") final String eventSourceName) {
+            return new NumberOfFailedEventsResponse(failedEventReplayer.getTotalNumberOfFailedEvents(eventSourceName));
+        }
+
+        @RequestMapping(
+            value = "/{event_source_name:.+}", method = RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE
+        )
+        public ResponseEntity<?> replay(@PathVariable("event_source_name") final String eventSourceName,
+                @RequestParam("number_of_failed_events") final long numberOfFailedEvents,
+                @RequestParam(value = "break_processing_on_exception", defaultValue = "false") final boolean breakProcessingOnException) {
+
+            try {
+                failedEventReplayer.replay(eventSourceName, numberOfFailedEvents, breakProcessingOnException);
+                return ResponseEntity.ok().build();
+            } catch (Exception ex) {
+                return ResponseEntity.badRequest().body(ex);
+            }
+        }
+
+        public DeferredResult<ResponseEntity<?>> replay(final String eventSourceName, final String eventId) {
```

Comment:
This class is a controller. These overloaded methods don't have a `@RequestMapping` anyways and can be removed.